model/atmosphere/advection/src/icon4py/model/atmosphere/advection/stencils/prepare_numerical_quadrature_list_for_cubic_reconstruction.py

```diff
@@ -0,0 +1,500 @@
+            )
+            - (
+                z_eta_4_1
+                * (astype(p_coords_dreg_v_2_y, wpfloat) - astype(p_coords_dreg_v_1_y, wpfloat))
+                + z_eta_4_2
+                * (astype(p_coords_dreg_v_3_y, wpfloat) - astype(p_coords_dreg_v_4_y, wpfloat))
+            )
+            * (
+                z_eta_4_3
+                * (astype(p_coords_dreg_v_4_x, wpfloat) - astype(p_coords_dreg_v_1_x, wpfloat))
+                - z_eta_4_4
+                * (astype(p_coords_dreg_v_2_x, wpfloat) - astype(p_coords_dreg_v_3_x, wpfloat))
+            )
+        ),
+        0.0,
+    )
+
+    z_gauss_pts_1_x = (
+        shape_func_1_1 * astype(p_coords_dreg_v_1_x, wpfloat)
+        + shape_func_2_1 * astype(p_coords_dreg_v_2_x, wpfloat)
+        + shape_func_3_1 * astype(p_coords_dreg_v_3_x, wpfloat)
+        + shape_func_4_1 * astype(p_coords_dreg_v_4_x, wpfloat)
+    )
+    z_gauss_pts_1_y = (
+        shape_func_1_1 * astype(p_coords_dreg_v_1_y, wpfloat)
+        + shape_func_2_1 * astype(p_coords_dreg_v_2_y, wpfloat)
+        + shape_func_3_1 * astype(p_coords_dreg_v_3_y, wpfloat)
+        + shape_func_4_1 * astype(p_coords_dreg_v_4_y, wpfloat)
+    )
+    z_gauss_pts_2_x = (
+        shape_func_1_2 * astype(p_coords_dreg_v_1_x, wpfloat)
+        + shape_func_2_2 * astype(p_coords_dreg_v_2_x, wpfloat)
+        + shape_func_3_2 * astype(p_coords_dreg_v_3_x, wpfloat)
+        + shape_func_4_2 * astype(p_coords_dreg_v_4_x, wpfloat)
+    )
+    z_gauss_pts_2_y = (
+        shape_func_1_2 * astype(p_coords_dreg_v_1_y, wpfloat)
+        + shape_func_2_2 * astype(p_coords_dreg_v_2_y, wpfloat)
+        + shape_func_3_2 * astype(p_coords_dreg_v_3_y, wpfloat)
+        + shape_func_4_2 * astype(p_coords_dreg_v_4_y, wpfloat)
+    )
+    z_gauss_pts_3_x = (
+        shape_func_1_3 * astype(p_coords_dreg_v_1_x, wpfloat)
+        + shape_func_2_3 * astype(p_coords_dreg_v_2_x, wpfloat)
+        + shape_func_3_3 * astype(p_coords_dreg_v_3_x, wpfloat)
+        + shape_func_4_3 * astype(p_coords_dreg_v_4_x, wpfloat)
+    )
+    z_gauss_pts_3_y = (
+        shape_func_1_3 * astype(p_coords_dreg_v_1_y, wpfloat)
+        + shape_func_2_3 * astype(p_coords_dreg_v_2_y, wpfloat)
+        + shape_func_3_3 * astype(p_coords_dreg_v_3_y, wpfloat)
+        + shape_func_4_3 * astype(p_coords_dreg_v_4_y, wpfloat)
+    )
+    z_gauss_pts_4_x = (
+        shape_func_1_4 * astype(p_coords_dreg_v_1_x, wpfloat)
+        + shape_func_2_4 * astype(p_coords_dreg_v_2_x, wpfloat)
+        + shape_func_3_4 * astype(p_coords_dreg_v_3_x, wpfloat)
+        + shape_func_4_4 * astype(p_coords_dreg_v_4_x, wpfloat)
+    )
+    z_gauss_pts_4_y = (
+        shape_func_1_4 * astype(p_coords_dreg_v_1_y, wpfloat)
+        + shape_func_2_4 * astype(p_coords_dreg_v_2_y, wpfloat)
+        + shape_func_3_4 * astype(p_coords_dreg_v_3_y, wpfloat)
+        + shape_func_4_4 * astype(p_coords_dreg_v_4_y, wpfloat)
+    )
+
```

Comment:
there are way too many type casts here, please check where they are needed and remove them where the float type is inferred from other fields

Reply:
This level of type cast is not required based on the written code. We will look at this tomorrow together.